### PR TITLE
Add query parameter support for Pilet razor pages

### PR DESCRIPTION
**Pull request by ghost**
# New Pull Request

## Prerequisites

Please make sure you can check the following boxes:

- [ ] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

## Type(s) of Changes

Added Atrribute and Value-Provider to process query parameters.

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

- Bug fix: Test solution was not able to build due missing project reference to Piral.BlazorUtils lib. Reference added.

- Feature
  - QueryParameterValueProvider added.
    - This Service inject the query parameter value to corresponding page instance.
  - PiralQueryParameterAttribute added
    - Marks a property which can be filled by a query parameter value.
  

### Remarks

I need to test this feature but I don't know how to test the updated Piral.Blazor.Core library.
In addition I'm not sure if it the perfect place to inject the values.

### CODE REVIEW

File: tests/Piral.Blazor.Tests.sln [hunk 1]
```diff
@@ -13,6 +13,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Piral.Blazor.SmokeTests.Pil
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Piral.Blazor.SmokeTests.PiletB", "Piral.Blazor.SmokeTests.PiletB\Piral.Blazor.SmokeTests.PiletB.csproj", "{EE84FF14-E742-475A-AA87-C800CFEDDDFF}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Piral.Blazor.Utils", "..\src\Piral.Blazor.Utils\Piral.Blazor.Utils.csproj", "{003F947A-B6A2-49AF-BCF9-05697CC1E030}"
```
Comment (FlorianRappl) [inline tests/Piral.Blazor.Tests.sln:16]: Any reason for adding this?

File: src/Piral.Blazor.Core/Program.cs [hunk 2]
```diff
@@ -24,7 +24,8 @@ public static async Task Main(string[] args)
             .AddSingleton(new HttpClient { BaseAddress = baseAddress })
             .AddSingleton<IComponentActivator, PiletComponentActivator>()
             .AddSingleton<IComponentActivationService, ComponentActivationService>()
-            .AddSingleton<IModuleContainerService, ModuleContainerService>();
+            .AddSingleton<IModuleContainerService, ModuleContainerService>()
+            .AddSingleton<IQueryParameterValueProvider, QueryParameterValueProvider>();
```
Comment (FlorianRappl) [inline src/Piral.Blazor.Core/Program.cs:28]: It's a viable approach, but why don't we just use the existing approach and extend it? Right now we already wrap / infer the parameters. We can just extend that functionality to look for the `PiralQueryParameterAttribute` and apply the correct value when needed.

Maybe I am missing something here. Thanks already!

Comment (ghost) [reply src/Piral.Blazor.Core/Program.cs:28]: I would like to extend the existing code. Unfortunately, I have no idea what is passed in the IDictionary<string, JsonElement>. 

How can I debug the core lib code?

I would like to implement it cleanly in the existing way.

File: tests/Piral.Blazor.Tests.sln (hunk 1 above)
Comment (ghost) [reply tests/Piral.Blazor.Tests.sln:16]: Piral.Blazor.SmokeTests got a reference to Piral.Blazor.Utils, but it is outside of the solution scope.
In a clean checkout Piral.Blazor.Utils will not be builded and tests can't be run.

Comment (FlorianRappl) [reply tests/Piral.Blazor.Tests.sln:16]: Well, yes, build comes before test - as the CI/CD pipeline does (which also runs the tests).

We can keep it but honestly I'd prefer if tests is actually just that.

File: src/Piral.Blazor.Core/Program.cs (hunk 2 above)
Comment (FlorianRappl) [reply src/Piral.Blazor.Core/Program.cs:28]: I've added a proposal to the branch. I use part of your code, but actually within the `Extensions` / where the usual `AdjustArguments` method is found:

https://github.com/smapiot/Piral.Blazor/blob/blazor-7.0/src/Piral.Blazor.Core/Extensions.cs#L144

Long story short; it seems to work fine that way and does not require an additional service. It essentially only triggers in absence of the standard parameter (i.e., no conflict potential here) and evaluates the mapping with your algorithm.

In my tests this seems to work as it should.

It is released in the current / latest preview.

Comment (ghost) [reply src/Piral.Blazor.Core/Program.cs:28]: That sounds very good.
I'm looking forward to the next release. 
Thank you very much   😀

**Pull request closed without merging**